Dockerfile

```
@@ -10,7 +10,7 @@ RUN go mod download
 # copy the rest of the sources and build
 COPY . .
 ARG GIT_TAG GIT_COMMIT TARGETARCH CGO_ENABLED
-RUN make build GIT_TAG="${GIT_TAG}" GIT_COMMIT="${GIT_COMMIT}" GO_BUILD_ENV="GOARCH=${TARGETARCH} CGO_ENABLED=${CGO_ENABLED}"
+RUN make build GIT_TAG="${GIT_TAG}" GIT_COMMIT="${GIT_COMMIT}" GO_BUILD_ENV='GOARCH=${TARGETARCH} CGO_ENABLED=${CGO_ENABLED}'
```

```suggestion
ARG GIT_TAG GIT_COMMIT TARGETARCH CGO_ENABLED=0
RUN make build GIT_TAG=${GIT_TAG} GIT_COMMIT=${GIT_COMMIT} GOARCH=${TARGETARCH} CGO_ENABLED=${CGO_ENABLED}
```

I don’t think we need `GO_BUILD_ENV` here at all. However, `GIT_TAG` and `GIT_COMMIT` are required build arguments – omitting them will result in "gitVersion": "" and "gitCommit": "" (this fallback behavior comes from https://github.com/kubernetes-sigs/kueue/pull/5210). Regarding `CGO_ENABLED`, I’d suggest setting a default value in the Dockerfile.